File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low = 0;
        int res = 0;
        int high = height.length - 1;

        while (low < high) {
            int curr = (high - low) * Math.min(height[low], height[high]);
            res = Math.max(res, curr);
            if (height[low] < height[high]) {
                low++;
            } else {
                high--;
            }
        }
        
        return res;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        res = 0

        while left != right:
            currArea = min(height[left], height[right]) * abs(right - left)
            res = max(currArea, res)
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        
        return res